webapp/channels/src/components/dialog_router/interactive_dialog_adapter.tsx

```diff
@@ -0,0 +1,844 @@
+                if (element.min_length > element.max_length) {
+                    errors.push({
+                        field: `${fieldPrefix}.min_length`,
+                        message: 'min_length cannot be greater than max_length',
+                        code: 'INVALID_FORMAT',
+                    });
+                }
+            }
+
+            // Validate against server limits
+            const maxLengthLimit = element.type === 'textarea' ? 3000 : 150;
+            if (element.max_length !== undefined && element.max_length > maxLengthLimit) {
+                errors.push({
+                    field: `${fieldPrefix}.max_length`,
+                    message: `max_length too large: ${element.max_length} > ${maxLengthLimit} (server limit for ${element.type})`,
+                    code: 'TOO_LONG',
+                });
+            }
+        }
+
+        // Validation for select fields
+        if (element.type === 'select') {
+            // Warn if both options and data_source are provided
+            if (element.options && element.data_source) {
+                errors.push({
+                    field: `${fieldPrefix}.options`,
+                    message: 'Select element cannot have both options and data_source',
+                    code: 'INVALID_FORMAT',
+                });
+            }
+
+            // Validate max_length for select fields (server limit is 3000)
+            if (element.max_length !== undefined && element.max_length > 3000) {
+                errors.push({
+                    field: `${fieldPrefix}.max_length`,
+                    message: `max_length too large: ${element.max_length} > 3000 (server limit for select)`,
+                    code: 'TOO_LONG',
+                });
+            }
+        }
+
+        // Validation for bool fields
+        if (element.type === 'bool') {
+            // Validate max_length for bool fields (server limit is 150)
+            if (element.max_length !== undefined && element.max_length > 150) {
+                errors.push({
+                    field: `${fieldPrefix}.max_length`,
+                    message: `max_length too large: ${element.max_length} > 150 (server limit for bool)`,
+                    code: 'TOO_LONG',
+                });
+            }
+        }
+
+        return errors;
+    };
+
+    /**
+     * Sanitize string input to prevent XSS attacks
+     */
+    private sanitizeString = (input: unknown): string => {
+        if (!this.conversionContext.sanitizeStrings) {
+            return String(input);
+        }
+
+        const str = String(input);
+
+        // Remove script tags and other potentially dangerous content
+        return str.
+            replace(/<script\b[^<]*(?:(?!<\/script>)<[^<]*)*<\/script>/gi, '').
+            replace(/<iframe\b[^<]*(?:(?!<\/iframe>)<[^<]*)*<\/iframe>/gi, '').
+            replace(/javascript:/gi, '').
+            replace(/on\w+\s*=/gi, ''); // Remove event handlers like onclick=
```

Comment:
This seems the kind of sanitization that can be cheated. Has this sanitization been validated by security?